### PR TITLE
dotnetCorePackages.sdk_5_0: init at version 5.0.100-preview.8.20417.9

### PR DESCRIPTION
**Pull request by JamieMagee**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/dotnet/default.nix [hunk 1]
```diff
@@ -102,4 +102,15 @@ rec {
...
+
+  # v5.0.0
+
+  sdk_5_0 = buildNetCoreSdk {
+    version = "5.0.100-preview.8.20417.9";
+    sha512 = {
+      x86_64-linux = "2a4m4jb75hc0l135pjpinawlc0jr3rz9hyfm22kkaqs5wacwj5d87lsds2wvp2jf7r1f445098sg65l6az9zqwg4da52qrvcv1ck1hd";
+      aarch64-linux = "1zb5f3ihlcjbij6sa9nfvj9rn4frhdbiw0nplqm8lf8q2z2y7h210p8r4l1qwkkm080185kj7kgd2hja54hddgghva26jpq6r36b7hj";
+      x86_64-darwin = "0c5cj6b0kc4nrsbxfmal4c1z5anqbxr6gd6knpqfb6m30z37qiks16xakacrm69313wmqpr9mmyiwpj38gcsw2dam9hgzawibwnv4fv";
+    };
```
Comment (felschr) [inline pkgs/development/compilers/dotnet/default.nix:114]:
```suggestion
    version = "5.0.100-rc.1.20452.10";
    sha512 = {
      x86_64-linux = "d7e709dacc4bb188c2380060d24bfb5b791240dc33af8499fb4a31e1885a9377dad1d1ebc76847432ea67d5e4ac832a31679dc293e09fa6dade28f5fbbe4db9b";
      aarch64-linux = "2d04890c71e845d1eb08f5dfbbb9c93024d7a52fb1cc3fd50bd51bc6bd44e455c5c82abc8f04eef23bd012984ae5f86143c600ceb49c4c733935d95d5b68785f";
      x86_64-darwin = "06bb40273071f3dd1e84ebf58abc7798795d5f1ac298f24bf7109d1597fd52ff31bcbf2b81f86d91d37ae293678d07f8da0469d7cbd318d19a8d718b6629dcac";
    };
```

**Review by felschr — COMMENTED** [review]
I'm not entirely sure about this but there seems generally no reason against including preview releases in nixpkgs.
Considering that .NET 5 is in the final stages of development personally I'd approve.

**Comment by JamieMagee** [comment]
Thanks. Yeah, I know it's close to GA.

I've missed the 20.09 cutoff, but I figure it'll be GA in time for 21.03, and it does no harm being in unstable in the meantime.

**Review by felschr — APPROVED** [review]
##### Reviewed points

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux

##### Comments

PR title should be updated.

**Comment by AndersonTorres** [comment]
Is there someone allocated for dotnet infrastructure?

**Comment by JamieMagee** [comment]
@AndersonTorres From the git history it looks like @jonringer & @nyanloutre are the best people to review these PRs in future.

**Comment by jonringer** [comment]
Yea, I've been kind of following the dotnet scene...

I'm using it again for work. So I might take some time to "make it nice" to use.

**Comment by jonringer** [comment]
also, IIRC, dotnet 5 is supposed to unify all of the various dotnet frameworks. If a stable release of dotnet5 comes out, we might want to make it the default

**Pull request merged**